Multiple Bug Fixes and Improvements Split Into Small Commits

JustinTimperio opened the pull request.

lu4p commented on torat_server/shell.go:
```diff
@@ -71,20 +70,24 @@ func Shell() {
+
 			if err != nil {
-				log.Fatalln("Could not change directory:", err)
+				fmt.Println(green("[Server] ") + red("[!] Cannot navigate to that path!"))
```
Please also print the error, as it might supply additional useful information.

lu4p replied:
Also return in the if so you don't need the else.

lu4p commented on torat_server/shell.go:
```diff
@@ -27,10 +24,10 @@ func Shell() {
 	cwd, err := os.Getwd()
 	if err != nil {
 		panic(err)
+	} else {
```
This else is useless, panic exits before anyway.

JustinTimperio replied:
I see this as a difference between writing styles. By making the error a switch it makes the logic more contained and easier for me to read. The compiler also will compile this, in the same way, give the surrounding code. 

I can change this if you feel strongly about it, but you may see this style by me elsewhere by force of habit.

lu4p replied:
Yes I feel strongly about that, because it reduces nesting.

JustinTimperio replied:
I'm not sure I understand your reasoning but you will prob see me doing this elsewhere. At this point, though I just want to get even 50% of the features working. We can clean up the little stuff as we go

lu4p reviewed (changes requested):
Looks okay just needs some minor changes.

JustinTimperio commented:
Okay, so I think I have made all the adjustments you wanted. I have also made some very large improvements to the ishell in both function and code structure. 

1. Fix Issue #172 
2. Fix Issue #174 
3. Fixed loss of ishell context during `cd` (IE: `[Server]` or `[Client]`)
4. Added Client ishell context (IE: `[Client.Name]`)
5. Improved Error Logging (Easier to read)
6. Split`torat_server/shell.go` into `torat_server/ishell_server.go` and `torat_server/ishell_client.go`. This is much easier to read and makes much sense structurally. This also prevents the conflict between the two `runCommand` functions (client vs server) which eliminates the need for `runcmd.go`

lu4p commented on torat_server/server_ishell.go:
```diff
@@ -0,0 +1,158 @@
+}
+
+// GET HOSTNAME OF CLIENT
+// TODO: Is this used? Where?
```
Yes here https://github.com/lu4p/ToRat/blob/78547ef9fc7bd55991e7163fc385376955270356/torat_server/server.go#L82, remove the todo.

lu4p commented on torat_server/server_ishell.go:
```diff
@@ -0,0 +1,158 @@
+package server
```
move this back to the other file

JustinTimperio replied:
move what back?

lu4p replied:
See the other comment

JustinTimperio commented on torat_server/server_ishell.go:
```diff
@@ -0,0 +1,158 @@
+			name := c.ReadLine()
+			client := activeClients[choice]
+
+			// ISSUE #173
```
No this is just me marking where the issue is. I'm not sure how to fix it yet

lu4p replied:
Then make this a todo ;)

JustinTimperio commented on torat_server/client_ishell.go:
```diff
@@ -212,78 +108,66 @@ func (client *activeClient) ls(c *ishell.Context) {
 	err = ioutil.WriteFile(path, r.Content, 0600)
 	if err != nil {
-		c.ProgressBar().Final(red("[!] Download failed:", err))
+		c.ProgressBar().Final(green("[Server] ") + red("[!] Download failed to write to path!"))
```
Because the error is coming from the server's file system. It gives context to what system has the path issue.

lu4p replied:
Ok thats actually a good idea. 

This it better because it also includes what exactly went wrong.
```go
c.ProgressBar().Final(green("@Server ") + red("[!] Download failed", err))
```

JustinTimperio commented on torat_server/server_ishell.go:
```diff
@@ -0,0 +1,158 @@
+	} else if cmdout == nil {
+		c.Println(green("[Server] ") + "No output returned by command!")
+		return
+	} else {
```
told you would see this XD

JustinTimperio commented on torat_server/server_ishell.go:
```diff
@@ -0,0 +1,158 @@
+
+// SERVER SIDE SHELL COMMAND
+func runCommand(c *ishell.Context) {
+	var osshell string
```
Not sure what that means

lu4p replied:
Like this
```go
var (
   osshell string
   osshellargs string
)
```

JustinTimperio commented on torat_server/client_ishell.go:
```diff
@@ -1,158 +1,49 @@
-var void int
-
-// Shell interactive shell
-func Shell() {
```
Again not sure what this means

lu4p replied:
The code you moved to the server_ishell.go file, move it back to this file.
